opencti-platform/opencti-front/src/public/components/ResetPassword.tsx

```diff
@@ -0,0 +1,336 @@
+import { useTheme } from '@mui/styles';
+import { Theme } from '@mui/material/styles/createTheme';
+import { useFormatter } from '../../components/i18n';
+import useApiMutation from '../../utils/hooks/useApiMutation';
+import { handleErrorInForm } from '../../relay/environment';
+import OTPForm from './OTPForm';
+import { ResetPasswordVerifyOtpMutation, ResetPasswordVerifyOtpMutation$data } from './__generated__/ResetPasswordVerifyOtpMutation.graphql';
+import { ResetPasswordAskSendOtpMutation } from './__generated__/ResetPasswordAskSendOtpMutation.graphql';
+import { ResetPasswordChangePasswordMutation } from './__generated__/ResetPasswordChangePasswordMutation.graphql';
+
+interface ResetProps {
+  onCancel: () => void;
+}
+
+export const AskSendOtpMutation = graphql`
+mutation ResetPasswordAskSendOtpMutation($input: AskSendOtpInput!){
+  askSendOtp(input: $input)
+}
+`;
+
+export const VerifyOtpMutation = graphql`
+mutation ResetPasswordVerifyOtpMutation($input: VerifyOtpInput!){
+  verifyOtp(input: $input) {
+    otp_activated
+  }
+}
+`;
+
+export const ChangePasswordMutation = graphql`
+mutation ResetPasswordChangePasswordMutation($input: ChangePasswordInput!){
+  changePassword(input: $input)
+}
+`;
+
+const resetValidation = (t: (v: string) => string) => Yup.object().shape({
+  email: Yup.string()
+    .required(t('This field is required'))
+    .email(t('The value must be an email address')),
+});
+
+const otpValidation = (t: (v: string) => string) => Yup.object().shape({
+  otp: Yup.string().required(t('This field is required')),
+});
+
+const passwordValidation = (t: (v: string) => string) => Yup.object().shape({
+  password: Yup.string().required(t('This field is required')),
+  password_validation: Yup.string()
+    .oneOf([Yup.ref('password'), undefined], t('The values do not match'))
+    .required(t('This field is required')),
+});
+
+interface ResetFormValues {
+  email: string;
+}
+
+interface ValidateTokenFormValues {
+  otp: string;
+}
+
+interface ResetPasswordFormValues {
+  password: string;
+  password_validation: string;
+}
+
+const STEP_ASK_RESET = 'ask';
+const STEP_VALIDATE_OTP = 'validate';
+const STEP_2FA = '2fa';
+const STEP_RESET_PASSWORD = 'reset';
+const FLASH_COOKIE = 'opencti_flash';
+
+const ResetPassword: FunctionComponent<ResetProps> = ({ onCancel }) => {
+  const theme = useTheme<Theme>();
+  const { t_i18n } = useFormatter();
+  const [step, setStep] = useState(STEP_ASK_RESET);
+  const [cookies, , removeCookie] = useCookies([FLASH_COOKIE]);
+  const [email, setEmail] = useState('');
+  const [otpError, setOtpError] = useState(false);
+  const [, setResendOtp] = useState(false);
+  const flashError = cookies[FLASH_COOKIE] || '';
+  removeCookie(FLASH_COOKIE);
+
+  const [askSentOtpCommitMutation] = useApiMutation<ResetPasswordAskSendOtpMutation>(
+    AskSendOtpMutation,
+    undefined,
+    {
+      successMessage: t_i18n('If your e-mail address is correct, an e-mail will be sent to you.'),
```

Comment:
correct or in our database ? Maybe something like:
```suggestion
      successMessage: t_i18n('If your e-mail address is found, an e-mail will be sent to you.'),
```